addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -388,13 +393,14 @@ def _rtc_leave_call(self, session_id=None):
             self.rtc_session_ids.unlink()
         else:
             self.channel_id._rtc_cancel_invitations(member_ids=self.ids)
-        if not self.channel_id.rtc_session_ids and self.channel_id.last_call_message_id:
-            # deducing information about the end of call through write_date
-            self.channel_id._message_update_content(
-                self.channel_id.last_call_message_id,
-                self.channel_id.last_call_message_id.body,
-                strict=False
+        if not self.channel_id.rtc_session_ids:
+            # sudo - dicuss.rtc.call.history: setting the end date of the call
+            # after it ends is allowed.
+            call_history_sudo = self.env["discuss.call.history"].sudo().search(
+                [("channel_id", "=", self.channel_id.id)], order="id DESC", limit=1
```

Comment:
For sorting it's best to use business fields rather than `id` when they are available:
`id` are technical details with no semantics, and their ordering is particularly weak to db imports/exports. Their ordering is also weak in regards to when they are acquired vs when the transaction is committed.

So `create_date` (or suggested `start_dt`) in this case (with potential index on it). `id` can (should) be kept as last resort to add determinism (when the other criteria is not unique).